CMakeLists.txt

```diff
@@ -50,10 +143,23 @@ if (OPENSSL_FOUND)
 else ()
     message( FATAL_ERROR "OpenSSL was not found. Please install OpenSSL" )
 endif (OPENSSL_FOUND)
+## Download Facebook Folly and build locally
+
+
+if (BUILD_LOCAL_DEPENDENCIES)
+	download_folly(${CMAKE_CURRENT_SOURCE_DIR} ${CMAKE_CURRENT_BINARY_DIR})
+	download_wangle(${CMAKE_CURRENT_SOURCE_DIR} ${CMAKE_CURRENT_BINARY_DIR})
+	download_zookeeper(${CMAKE_CURRENT_SOURCE_DIR} ${CMAKE_CURRENT_BINARY_DIR})
+endif(BUILD_LOCAL_DEPENDENCIES)
+
 # ensure we have required dependencies
 find_package(Threads)
-find_package(Boost REQUIRED COMPONENTS thread system filesystem)
+find_package(Boost REQUIRED COMPONENTS context thread system filesystem regex)
+find_package(LibEvent REQUIRED)
 find_package(Gflags REQUIRED)
+if (BUILD_LOCAL_DEPENDENCIES)
+	find_package(Sodium REQUIRED)
```

Comment:
not sure I followed this, we need it irrespective of whether the dependencies are downloaded right? (for folly)